test/lib/flat_mutation_reader_assertions.hh

```diff
@@ -760,10 +819,15 @@ public:
         if (!mfopt->is_end_of_partition()) {
             BOOST_FAIL(format("Expected partition end but got {}", mutation_fragment_v2::printer(*_reader.schema(), *mfopt)));
         }
+        reset_rt();
```

Comment:
This shouldn't be needed. A valid v2 stream always has a null active range tombstone when you get to end-of-partition. In fact we should have an assert here checking this: `BOOST_REQUIRE(!_rt);`.
The only situation where this is not the case is when the user stops consuming the content of a stream abruptly and calls `fast_forward_to()` or `next_partition()`. But in either of these cases, `produces_partition_end()` will for sure not be called, but `produces_partition_start()` may, so we can and should keep `reset_rt()` there.